File: iossdkdemo/SDKDemo/APIControllers/SurveyListViewController.m
//
// Copyright (c) 2016 OnePoint Global Ltd. All rights reserved.
//
// This code is licensed under the OnePoint Global License.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
// THE SOFTWARE.
//
#import "SurveyListViewController.h"
#import "SurveyTableViewCell.h"
#import "TakeSurveyViewController.h"
#import <OPGSDK/OPGSDK.h>
#import <OPGSDK/OPGSurvey.h>


@interface SurveyListViewController ()

@property(strong, nonatomic) NSString* surveyRef;

@end

@implementation SurveyListViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.surveyTableView.separatorColor = [UIColor orangeColor];
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    
}

#pragma mark - TableView Delegate Methods
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.surveyListArray.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    SurveyTableViewCell* tableViewCell = [tableView dequeueReusableCellWithIdentifier:@"tableViewCell"];
    
    /* OPGSurvey class has properties such as Name, description, IsOffline, LastUpdatedDate, IsGeoFencing, ScriptID and SurveyReference
     Getting instance of OPGSurvey from the array */
    OPGSurvey *survey = [self.surveyListArray objectAtIndex:indexPath.row];
    
    [tableViewCell fillCell:survey];         // filling the custom tableview cell with OnePoint Survey class Instance
    
    return tableViewCell;
}

-(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    OPGSurvey *survey = [self.surveyListArray objectAtIndex:indexPath.row];
    self.surveyRef = survey.surveyReference;  // Survey reference property of OPGSurvey class is passed to next controller which is required to run the survey.
    [self performSegueWithIdentifier:@"embedTakeSurvey" sender:self];

}

-(CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
    return 100.0f;
}

-(CGFloat)tableView:(UITableView *)tableView heightForFooterInSection:(NSInteger)section {
    return 3.0f;
}

#pragma mark - Segues Operations
-(void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    if ([[segue identifier] isEqualToString:@"embedTakeSurvey"]) {
        // Get TakeSurveyViewController view
        TakeSurveyViewController *viewController = [segue destinationViewController];
        viewController.surveyRef = self.surveyRef;
    }
    
}


@end

File: iossdkdemo/SDKDemo-Swift/APIControllers-Swift/SurveyListViewController.swift
//
// Copyright (c) 2016 OnePoint Global Ltd. All rights reserved.
//
// This code is licensed under the OnePoint Global License.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
// THE SOFTWARE.
//

import UIKit
import Foundation



class SurveyListViewController : UIViewController, UITableViewDataSource, UITableViewDelegate
{
    @IBOutlet weak var surveyListTableView : UITableView!
    var surveyListArray : NSArray!                        // Array of OPGSurvey Class from Previous ViewController
    var surveyReference : NSString!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        self.surveyListTableView.separatorColor = UIColor.orange
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return self.surveyListArray.count
    }
    
    func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let tableViewCell : SurveyTableViewCell = tableView.dequeueReusableCell(withIdentifier: "tableViewCell") as! SurveyTableViewCell
        /* OPGSurvey class has properties such as Name, description, IsOffline, LastUpdatedDate, IsGeoFencing, ScriptID and SurveyReference
         Getting instance of OPGSurvey from the array */
        let survey : OPGSurvey = surveyListArray.object(at: (indexPath as NSIndexPath).row) as! OPGSurvey
        tableViewCell.fillCell(survey)                // filling the custom tableview cell with OnePoint Survey class Instance
        return tableViewCell 
    }
    
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return 100.0
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath)
    {
        let survey : OPGSurvey = self.surveyListArray.object(at: (indexPath as NSIndexPath).row) as! OPGSurvey
        self.surveyReference = survey.surveyReference as NSString!    // Survey reference property of OPGSurvey class is passed to next controller which is required to run the survey.
        self.performSegue(withIdentifier: "embedTakeSurvey", sender: self)
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if(segue.identifier == "embedTakeSurvey")
        {
            // Get TakeSurveyViewController view
            let viewController : TakeSurveyViewController = segue.destination as! TakeSurveyViewController
            viewController.surveyRef=self.surveyReference as String
        }
        
    }
    
}
